### ISSUE TITLE
related languages

### ISSUE THREAD

**Issue by relvao** [json-api/json-api#613]
I need to identify related resources in different languages. I'm not sure how it would be done. Any ideas?

We consider that an article in 2 different locales is in fact 2 different documents that have a relationship. They can have different properties, eg. diferrent tags. 

Returning all the data in the same request wouldn't be an option. Furtermore we have several versions of each document, up to 100, as we have several versions in the same language depending on the country. 

Example URLs:
/en/GB/articles/my-test
/pt/PT/artigos/o-meu-teste

In HTML we have rel="alternate" hreflang="x". Google advices the use of the link header for non html pages - Link: http://es.example.com/; rel="alternate"; hreflang="es" but I would rather have that information in the response body. Do we have something comparable in the spec?

Thanks, 
Tiago

**Comment by gr0uch** [comment]
According to HTTP semantics, the `Accept-Language` header should be used to negotiate the language of the response, and `Content-Language` should indicate the language of the response.
http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

**Comment by relvao** [comment]
Hi @daliwali 

Thanks for the input. We are aware of those headers and content-language is properly applied in our API. Due to a combination of factors we chose to use the url to indicate the language being requested. 

The fact is we do have different URLs for localised versions of the page, as well as a lot of websites out there. Does anyone now how to best represent this using json-api?

TL;DR;
We chose to use a URL to identify the locale to make our API easier to consume.

From google:
"Keep the content for each language on separate URLs. Don’t use cookies to show translated versions of the page. Consider cross-linking each language version of a page. That way, a French user who lands on the German version of your page can get to the right language version with a single click.

Avoid automatic redirection based on the user’s perceived language. These redirections could prevent users (and search engines) from viewing all the versions of your site."

There is some information about the issue in these links:
- http://webmasters.stackexchange.com/questions/403/how-should-i-structure-my-urls-for-both-seo-and-localization
- https://support.google.com/webmasters/answer/182192?hl=en#2

**Comment by ethanresnick** [comment]
If your articles are different documents with, for example, different tags, then they can simply be different resources in the json-api sense, with different ids and URLs, but all part of the same `"articles"` collection. So it sounds like the only issue is how to link an article to its counterparts in another language. My suggestion would be to just add an `"alternates"` key to ~~`"links"`~~ `"relationships"` for this purpose.

So I'd have something like:

```
GET /en/GB/articles/my-test
Accept: application/vnd.api+json

{
  "data": {
    "type": "articles",
    "id": "en/GB/my-test",
    "attributes": {
      "language": "en-GB",
      //... more attributes
    },
    "relationships": {
      "alternates": {
        "data": [
          {"type": "articles", "id": "/pt/PT/o-meu-teste", "meta": {"language": "pt-PT"}}, 
          // more alternates 
        ],
        "related": "/en/GB/articles/my-test/alternates"
      }
      "tags": {
        // the article's tags
      }
  }
}
```

Note that I added the `"meta"` within each `"alternates"` linkage only so the client could tell the language of each alternate without having to request it first. That's entirely optional, but it seems useful.

**Comment by relvao** [comment]
@ethanresnick Thanks. Looks good.
